### PR TITLE
chore: Fix missing template parameter for IEventListener

### PR DESCRIPTION
**Pull request by come-nc**
* Resolves: # 

## Summary

Hopefully fixes all `MissingTemplateParam` psalm errors from baseline.

## Checklist

- Code is [properly formatted](https://docs.nextcloud.com/server/latest/developer_manual/digging_deeper/continuous_integration.html#linting)
- [Sign-off message](https://github.com/src-d/guide/blob/master/developer-community/fix-DCO.md) is added to all commits
- [x] Tests ([unit](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#unit-tests), [integration](https://docs.nextcloud.com/server/latest/developer_manual/app_development/tutorial.html#integration-tests), api and/or acceptance) are included
- [x] Screenshots before/after for front-end changes
- [x] Documentation ([manuals](https://github.com/nextcloud/documentation/) or wiki) has been updated or is not required
- [x] [Backports requested](https://github.com/nextcloud/backportbot/#usage) where applicable (ex: critical bugfixes)

### CODE REVIEW

**Review by ChristophWurst — APPROVED** [review]

**Review by st3iny — APPROVED** [review]
Epic

**Comment by come-nc** [comment]
```
72 errors fixed
```
:tada:

File: apps/twofactor_backupcodes/lib/AppInfo/Application.php [hunk 1]
```diff
@@ -56,8 +57,8 @@ public function register(IRegistrationContext $context): void {
 		$context->registerEventListener(CodesGenerated::class, ActivityPublisher::class);
 		$context->registerEventListener(CodesGenerated::class, RegistryUpdater::class);
 		$context->registerEventListener(CodesGenerated::class, ClearNotifications::class);
-		$context->registerEventListener(IRegistry::EVENT_PROVIDER_ENABLED, ProviderEnabled::class);
-		$context->registerEventListener(IRegistry::EVENT_PROVIDER_DISABLED, ProviderDisabled::class);
+		$context->registerEventListener(TwoFactorProviderForUserRegistered::class, ProviderEnabled::class);
+		$context->registerEventListener(TwoFactorProviderForUserUnregistered::class, ProviderDisabled::class);
```
Comment (nickvergessen) [inline apps/twofactor_backupcodes/lib/AppInfo/Application.php:61]: Failing unit tests are related?
```

There were 2 failures:

1) OCA\TwoFactorBackupCodes\Tests\Unit\Listener\ProviderDisabledTest::testHandleNoActiveProvider
Expectation failed for method name is "remove" when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.

2) OCA\TwoFactorBackupCodes\Tests\Unit\Listener\ProviderEnabledTest::testHandleCodesGeneratedEventNoBackupcodes
Expectation failed for method name is "add" when invoked 1 time(s).
Method was expected to be called 1 times, actually called 0 times.
```

Comment (nickvergessen) [reply apps/twofactor_backupcodes/lib/AppInfo/Application.php:61]: Works on master, fails on this branch

**Pull request merged**